sympy/algebras/quaternion.py

```diff
@@ -720,3 +720,388 @@ def to_rotation_matrix(self, v=None):
+        are real numbers then the vector part of q is b*i + c*j + d*k.
+
+        Returns
+        =======
+
+        Quaternion: representing vector part of the quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.vector_part()
+        0 + 1*i + 1*j + 1*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.vector_part()
+        0 + 8*i + 13*j + 12*k
+
+        See Also
+        ========
+        https://en.wikipedia.org/wiki/Versor
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d)
+
+    def axis(self):
+        """
+        Returns the axis of the quaternion.
+
+        Explanation
+        ===========
+
+        The axis of a quaternion is the normalized versor (vector part) of that quaternion.
+
+        Returns
+        =======
+
+        Quaternion: representing axis of the provided quaternion.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.axis()
+        0 + sqrt(3)/3*i + sqrt(3)/3*j + sqrt(3)/3*k
+
+        >>> q = Quaternion(4, 8, 13, 12)
+        >>> q.axis()
+        0 + 8*sqrt(377)/377*i + sqrt(377)/29*j + 12*sqrt(377)/377*k
+
+        See Also
+        ========
+        vector
+
+        """
+
+        return Quaternion(0, self.b, self.c, self.d).normalize()
+
+    def is_pure(self):
+        """
+        Returns true if the quaternion is pure else returns false.
+
+        Explanation
+        ===========
+
+        A pure quaternion (also a vector quaternion) is a quaternion with scalar
+        part equal to 0.
+
+        Examples
+        ========
+
+        >>> from sympy.algebras.quaternion import Quaternion
+        >>> q = Quaternion(1, 1, 1, 1)
+        >>> q.is_pure()
+        False
+
+        >>> q = Quaternion(0, 8, 13, 12)
+        >>> q.is_pure()
+        True
+
+        See Also
+        ========
+        scalar
+
+        """
+
+        if isinstance(self.a, Symbol):
+            z = self.a.is_zero
+            if not z:
+                return False
+            elif z == None:
+                return
+            else:
+                x = fuzzy_and(self.b.is_zero, self.c.is_zero,self.d.is_zero)
+                if x:
+                    return False
+                if x == None:
+                    return
+                else:
+                    return True
+        return self.a == 0  and not (self.a == self.b == self.c == self.d)
+
+    def angle(self):
+        """
+        Returns the angle of the quaternion measured in the real-axis plane.
+
+        Explanation
+        ===========
+
+        If q is a quaternion given by q = a + b*i + c*j + d*k where a, b, c and d
+        are real numbers then the angle of the quaternion is given by
+
+            angle = atan2(sqrt(b**2 + c**2 + d**2), a)
```

Comment:
How exactly does this render in the docs?

I think it might be better as LaTeX.